cmd/prometheus/main.go

```diff
@@ -94,13 +97,8 @@ func main() {
 
 	a.HelpFlag.Short('h')
 
-	a.Flag("log.level",
-		"Only log messages with the given severity or above. One of: [debug, info, warn, error, fatal]").
-		Default("info").StringVar(&cfg.logLevel)
-
-	a.Flag("log.format",
-		`Set the log target and format. Example: "logger:syslog?appname=bob&local=7" or "logger:stdout?json=true"`).
-		Default("logger:stderr").StringVar(&cfg.logFormat)
+	a.Flag(promlog.LevelFlagName, promlog.LevelFlagHelp).
```

Comment:
I disagree here. This would require our logging package to depend on our chosen flag package, which is very unnecessary.
Plugging basic things together in `main` is completely fine.

Those unnecessary cross-dependencies tend to propagate and you end up with all those packages importing each other.

Example: we switched to a different flag package in prometheus/prometheus. All other applications should to, but it will propagate gradually.